crates/hir-def/src/nameres/collector.rs

```diff
@@ -2141,26 +2142,34 @@ impl ModCollector<'_, '_> {
 
     fn collect_macro_call(&mut self, mac: &MacroCall, container: ItemContainerId) {
         let ast_id = AstIdWithPath::new(self.file_id(), mac.ast_id, ModPath::clone(&mac.path));
+        let db = self.def_collector.db;
 
-        // Case 1: try to resolve in legacy scope and expand macro_rules
+        // FIXME: Eagerly expanding in "Case 1" is insufficient since "Case 2" may also define
```

Comment:
Using the term eager might be a bit confusing here as there is a concept of eager vs lazy macros

Reply:
I've replaced it with "immediately" as used in the comment below.